src/DaysCalculator.java

```diff
@@ -0,0 +1,62 @@
+import java.util.Scanner;
+
+public class LeapYear {
+    public static void main(String[] args) {
+        int year, month;
+        int days;
+        Scanner s = new Scanner(System.in);
+
+        // Get input of the year
+        System.out.println("Enter a year after 1582: ");
+        do {
+            year = s.nextInt();
+            if (year <= 1582) {
+                System.out.println("Invalid year, please enter a year greater than 1582. Try again.");
+            }
+        } while (year <= 1582);
+
+        // Get input of the month
+        System.out.println("Enter a month: ");
+        do {
+            month = s.nextInt();
+            if (month < 1 || month > 12) {
+                System.out.println("Invalid month, please enter a valid month between 1 and 12. Try again.");
+            }
+        } while (month < 1 || month > 12);
+        s.close();
+
+        switch (month) {
```

Comment:
As the Task says: *Then, using the *switch expression*, initialize variable days*.

You *are not* using a "switch expression", but a "switch statement"